README.md

```diff
@@ -50,6 +50,22 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. `->` (arrow) expresses a function's explicit parameter and return value types. `String` -> `[String]` -> `[String]` is a function which takes a `String` and an `[String]` and returns an `[String]`.
+2. `~>` (squiggly arrow) expresses a function's implicit parameter type. When a function is a property of a type, it is called a method. All methods have an implicit parameter type - the type of which they are a property of.
+3. `=>` (fat arrow) expresses constraints on type variables. For example, `a -> b` is a function which takes an argument and returns a result. The argument can be any type. The result can be any type (including the same type as the argument). All this applies to the function described by `Functor f => f a -> f b` but we now require the type `f` to satisfy the Functor specification.
+
+i.e.
+
+```
+sequence :: Apply f, Traversable t => t (f a) ~> (b -> f b) -> f (t a)
+'------'    '--------------------'    '-----'    '-------------------'
+   '- method name     '- constraints     '- method target  '- method arguments
```

Comment:
Note: I changed the example to `map`'s definition, because in `traverse`'s:

```
traverse :: Applicative f, Traversable t => t a ~> (TypeRep f, a -> f b) -> f (t b)
```

I don't understand what `TypeRep` is. I know there's this section in the README (but I still don't get it): 

```
## Type representatives

Certain behaviours are defined from the perspective of a member of a type.
Other behaviours do not require a member. Thus certain algebras require a
type to provide a value-level representative (with certain properties). The
Identity type, for example, could provide `Id` as its type representative:
`Id :: TypeRep Identity`.

If a type provides a type representative, each member of the type must have
a `constructor` property which is a reference to the type representative.
```

Maybe someone can add an example?